monai/losses/dice.py

```
@@ -846,6 +849,17 @@ def forward(self, input: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
             else:
                 target = one_hot(target, num_classes=n_pred_ch)
 
+        # Apply activation before removing background to ensure softmax/sigmoid works correctly
+        if self.sigmoid:
+            input = torch.sigmoid(input)
+        elif self.softmax:
+            if n_pred_ch == 1:
+                warnings.warn("single channel prediction, `softmax=True` ignored.")
```

Ruff (0.15.18)

[warning] No explicit `stacklevel` keyword argument found

Set `stacklevel=2`

(B028)